torch/serialization.py

```diff
@@ -783,3 +788,11 @@ def persistent_load(saved_id):
     result = unpickler.load()
 
     return result
+
+
+def _is_torchscript_zip(zip_file):
+    for file_name in zip_file.get_all_records():
+        parts = file_name.split(os.sep)
+        if len(parts) > 1 and parts[1] == 'constants.pkl':
```

Comment:
How reliable is this in the future?
Can we add a comment where this file name is used in the jit save to say that this is used to detect torchscript zip files here?

Reply:
It should be fairly reliable, `constants.pkl` has been around for a while and there's really no compelling reason to replace it right now. The latest commit adds a test that asserts it exists and an explanation for why.

Reply:
Looks good, thanks.